packages/react-positioning/src/usePopper.ts

```diff
@@ -0,0 +1,401 @@
+import { useEventCallback, useIsomorphicLayoutEffect, useCallbackRef, useFirstMount } from '@fluentui/react-utilities';
+import { useFluent } from '@fluentui/react-provider';
+import {
+  isBrowser,
+  getScrollParent,
+  applyRtlToOffset,
+  getPlacement,
+  getReactFiberFromNode,
+  getBoundary,
+} from './utils/index';
+import * as PopperJs from '@popperjs/core';
+import * as React from 'react';
+
+import { PopperOptions } from './types';
+type PopperInstance = PopperJs.Instance & { isFirstRun?: boolean };
+
+//
+// Dev utils to detect if nodes have "autoFocus" props.
+//
+
+/**
+ * Detects if a passed HTML node has "autoFocus" prop on a React's fiber. Is needed as React handles autofocus behavior
+ * in React DOM and will not pass "autoFocus" to an actual HTML.
+ */
+function hasAutofocusProp(node: Node): boolean {
+  // https://github.com/facebook/react/blob/848bb2426e44606e0a55dfe44c7b3ece33772485/packages/react-dom/src/client/ReactDOMHostConfig.js#L157-L166
+  const isAutoFocusableElement =
+    node.nodeName === 'BUTTON' ||
+    node.nodeName === 'INPUT' ||
+    node.nodeName === 'SELECT' ||
+    node.nodeName === 'TEXTAREA';
+
+  if (isAutoFocusableElement) {
+    return !!getReactFiberFromNode(node)?.pendingProps.autoFocus;
+  }
+
+  return false;
+}
+
+function hasAutofocusFilter(node: Node) {
+  return hasAutofocusProp(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
+}
+
+/**
+ * Provides a callback to resolve Popper options, it's stable and should be used as a dependency to trigger updates
+ * of Popper options.
+ *
+ * A callback is used there intentionally as some of Popper.js modifiers require DOM nodes (targer, container, arrow)
+ * that can't be resolved properly during an initial rendering.
+ */
+function usePopperOptions(options: PopperOptions, popperOriginalPositionRef: React.MutableRefObject<string>) {
+  const {
+    autoSize,
+    flipBoundary,
+    offset,
+    onStateUpdate,
+    overflowBoundary,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    unstable_disableTether,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    unstable_pinned,
+  } = options;
+
+  const isRtl = useFluent().dir === 'rtl';
```

Comment:
_nit:_ I suggest to move out it to `usePopper()` hook, but as you wish as currently the scope is limited to `usePopperOptions`

Reply:
Let's go as it is for now 👍